tests/unit/style/properties/serialization.rs

```diff
@@ -576,29 +576,62 @@ mod shorthand_serialization {
+        use style::properties::longhands::font_stretch::SpecifiedValue as FontStretch;
+        use style::properties::longhands::font_style::SpecifiedValue as FontStyle;
+        use style::properties::longhands::font_variant::SpecifiedValue as FontVariant;
+        use style::properties::longhands::font_weight::SpecifiedValue as FontWeight;
         use style::properties::longhands::line_height::SpecifiedValue as LineHeight;
+        use super::*;
 
+        // Currently will fail unless product == "gecko" or "none".
+        // TODO: Remove ignore when we can test with those products set.
         #[test]
-        fn font_should_serialize_all_available_properties() {
+        #[ignore]
+        fn should_serialize_to_empty_if_there_are_nondefault_subproperties() {
+            // Font shorthand properties
+            let font_family = DeclaredValue::Value(
+                FamilyContainer(vec![FontFamily::Generic(Atom::from("serif"))])
+            );
+            let font_style = DeclaredValue::Value(FontStyle::italic);
+            let font_variant = DeclaredValue::Value(FontVariant::normal);
+            let font_weight = DeclaredValue::Value(FontWeight::Bolder);
+            let font_size = DeclaredValue::Value(FontSizeContainer(
+                LengthOrPercentage::Length(NoCalcLength::from_px(4f32)))
+            );
+            let font_stretch = DeclaredValue::Value(FontStretch::expanded);
+            let line_height = DeclaredValue::Value(LineHeight::Number(3f32));
+
+            // Properties not part of the font shorthand
+            let font_kerning = DeclaredValue::Value(FontKerning::none);
+            let language_override = DeclaredValue::Value(FontLanguageOverride::Normal);
+
             let mut properties = Vec::new();
+            properties.push(PropertyDeclaration::FontFamily(font_family));
+            properties.push(PropertyDeclaration::FontStyle(font_style));
+            properties.push(PropertyDeclaration::FontVariant(font_variant));
+            properties.push(PropertyDeclaration::FontWeight(font_weight));
+            properties.push(PropertyDeclaration::FontSize(font_size));
+            properties.push(PropertyDeclaration::FontStretch(font_stretch));
+            properties.push(PropertyDeclaration::LineHeight(line_height));
+            properties.push(PropertyDeclaration::FontLanguageOverride(language_override));
+            properties.push(PropertyDeclaration::FontKerning(font_kerning));
 
+            let serialization = shorthand_properties_to_string(properties);
+            assert_eq!(serialization, "font-language-override: normal; font-kerning: none;");
+        }
 
+        #[test]
+        fn font_should_serialize_all_available_properties() {
             let font_family = DeclaredValue::Value(
-                FamilyContainer(vec![FontFamily::Generic(atom!("serif"))])
+                FamilyContainer(vec![FontFamily::Generic(Atom::from("serif"))])
```

Comment:
@nox It looks like you removed this comment, but in case it's not clear to others now that I've addressed the other nits: This did not compile using `atom!`, because the`string_cache` crate is not available, but this variation works. This test was commented out before, so I just uncommented and got it to compile by using `Atom::from` based on a suggestion from someone in IRC.